aligner_seed.cpp

```diff
@@ -77,50 +77,55 @@ Constraint Constraint::editBased(int edits) {
 	return c;
 }
 
-// Input to seachSeedBi
-// local seed alignment cache and associated instatiated seed
-class CacheAndSeed {
-	public:
-		CacheAndSeed()
-		: seq(NULL), seq_len(0)  {}
-
-		CacheAndSeed(
-			const char *   _seq,             // sequence of the local seed alignment cache
-			const uint8_t  _seq_len          // and its length
-		)
-		: seq(NULL), seq_len(0)  // just set a default
-		{ reset(_seq, _seq_len); }
-
-		void reset(
-			const char *   _seq,             // sequence of the local seed alignment cache
-			const uint8_t  _seq_len          // and its length
-		)
-		{
-			seq = _seq;
-#ifndef NDEBUG
-			if (_seq_len>127) {printf("Unexpected seq_len %i\n",int(seq_len)); throw 1;}
-#endif
-			seq_len = _seq_len;
-		}
+class SeedAlignerSearchData {
+public:
+	// create an empty bwt
+	// and constratins from seed, for initial searchSeedBi invocation
+	SeedAlignerSearchData()
+	: bwt()
+	, sak()
+	, nlex(0)
+	, need_reporting(false)
+	{} // may need to initialize sak with NULL and 0
```

Comment:
You can remove the comment;
sak() initializer already does the proper initialization